(Feature) Allow min cap manage screen

fernandomg opened the pull request:
Closes #930 

I wasn't able to update `minCap` after crowdsale creation.

I tried with and without setting an initial minCap value in the crowdsale creation process.

Please @vbaranov review

---

Besides, @dennis00010011b `manage` screen layout has slightly changed, so it is possible that e2e test fails.

dennis00010011b commented:
4. Error message if   ```globalMinCap === "" ```

<img width="1440" alt="screen shot 2018-06-05 at 16 40 03" src="https://user-images.githubusercontent.com/14067096/41008357-7944743c-68df-11e8-9eea-de2190d1c60d.png">

dennis00010011b commented:
5. There is no validation  if minCap > totalSupply

fernandomg commented on src/components/manage/ManageDutchAuctionBlock.js:
```diff
@@ -30,22 +31,20 @@ export const ManageDutchAuctionBlock = ({
+      const tierHasStarted = !isDateLaterThan()(dateToTimestamp(initialStartTime))(Date.now())
       const tierHasEnded = !isDateLaterThan()(dateToTimestamp(initialEndTime))(Date.now())
-      const canEditWhiteList = canEditTiers && !tierHasEnded
+      const canEditWhiteList = canEditTiers && !tierHasEnded && !tierHasStarted
```
fixed

fernandomg commented:
@dennis00010011b @vbaranov fixes done, thanks!

Still cannot update minCap, did you guys were able to do it?

dennis00010011b commented:
@fernandomg  There is still issue, can't add whitelist from manage page. Decimals =18
<img width="1440" alt="screen shot 2018-06-06 at 08 54 31" src="https://user-images.githubusercontent.com/14067096/41050186-812dda02-6967-11e8-8d4b-58ff3242651d.png">

dennis00010011b commented:
2. There is no validation if minCap >totalSupply
3. In case if all tiers are whitelisted field minCap should be always disabled
4. minCap enabled if crowdsale not modifiable

Yes, I can't update minCap too

fernandomg commented:
> In case if all tiers are whitelisted field minCap should be always disabled

fixed

> can't add whitelist from manage page. Decimals =18

fixed

---

> There is no validation if minCap >totalSupply

If that's the case, the crowdsale creation process is wrong as well, because if you enable whitelists for only one tier, minCap is disabled and set to zero. 

> minCap enabled if crowdsale not modifiable

Thinking a bit further, rule should be: if all tiers without whitelist enabled are not modifiable, then minCap should not be modifiable either.

Please @dennis00010011b @vbaranov confirm the last two items.

dennis00010011b commented:
@fernandomg 
There is still issue:  minCap field disabled even if tier modifiable, no whitelisted

vbaranov commented:
@dennis00010011b @fernandomg 

I guess the rule on the management page could be the same as in step 3 of deployment process:

global min cap is disabled if any of a tier is whitelisted. If none of the tiers is whitelisted, global min cap is enabled and should be less than total crowdsale supply.

fernandomg commented:
@dennis00010011b @vbaranov 

`minCap` is editable only if there's no whitelist, and it cannot be greater than the minimum supply for the tiers.

dennis00010011b commented:
@fernandomg 
minCap is editable before start if crowdsale isn't modifiable

<img width="1440" alt="screen shot 2018-06-07 at 08 05 33" src="https://user-images.githubusercontent.com/14067096/41108629-e81390d2-6a29-11e8-8469-4519511e6c64.png">

fernandomg commented:
> minCap is editable before start if crowdsale isn't modifiable

@dennis00010011b Fixed, so if there's at least one editable tier `minCap` will be editable as well.
